### ISSUE TITLE
in progress items stacked incorrectly in menu

### ISSUE THREAD

**Issue by dev-dsp** [CleverRaven/Cataclysm-DDA#76460]
### Describe the bug

Different in progress items sometimes are stacked incorrectly, see screenshot

### Attach save file

N/A

### Steps to reproduce

1. Leave several unfinished crafts laying around
2. 'a'

### Expected behavior

In progress items stacked by target item

### Screenshots

![image](https://github.com/user-attachments/assets/30851af7-ece3-4131-94c1-b91489d46666)

### Versions and configuration

- OS: Linux
    - OS Version: LSB Version: n/a; Distributor ID: Gentoo; Description: Gentoo Linux; Release: 2.15; Codename: n/a; 
- Game Version: cdda-experimental-2024-09-15-0626 1925be0 [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Portal Storms Ignore NPCs [personal_portal_storms],
    Slowdown Fungal Growth [no_fungal_growth],
    Mind Over Matter [mindovermatter],
    Stats Through Kills [stats_through_kills],
    SpeedyDex [speedydex]
]

**Comment by PatrikLundell** [comment]
No, they shouldn't stack, because their progress isn't identical, it's only the rounded (or truncated?) progress indication that's the same.

**Comment by dev-dsp** [comment]
![image](https://github.com/user-attachments/assets/40688681-d02c-4db7-99e4-35856dfaa3e0)
